state-network.md

```diff
@@ -259,165 +247,72 @@ content_for_retrieval  := Container(code: ByteList)
-1:             B*                          C
-             /   \
-           /       \
-         /           \
-2:      D             E*
-                     / \
-                    /   \
-3:                 F     G*
-                        / \
-4:                     H*  I
-
-- "*" denotes trie node modified
-```
-
-In the example proof diagramed here we can see a *leaf* node `H` which
-represents the only modified leaf state in this proof.  Note that all of the
-nodes along the path leading to `H` are also modified.
-
-The *minimal* proof would contain the nodes `[D, F, H, I, C]`
-
-The bridge node would search the DHT for nodes that are *interested* in storing
-the node `H` and gossip this proof to those nodes.
-
-The recipients of this gossip are then responsible for gossiping the parent
-intermediate node `G`.  To do so, they would strip off the `H` and `I` nodes
-from this proof resulting in the following:
-
-```
-EXAMPLE: Recursive Gossip Round 1
-
-0:                           A*
-                            / \
-                          /     \
-                        /         \
-                      /             \
-                    /                 \
-                  /                     \
-                /                         \
-1:             B*                          C
-             /   \
-           /       \
-         /           \
-2:      D             E*
-                     / \
-                    /   \
-3:                 F     G*
-
-4:
-```
+At each block we construct a proof for each new or modified value in the trie, which is stored in
+the leaf nodes. The bridge node would search the DHT for nodes that are *interested* in storing
+each leaf node and gossip the proof to those nodes.
 
-At this stage, the minimal proof for `G` would be `[D, F, G, C]`.  The nodes
-which received the initial gossip message for `H` would construct this proof
-by removing the un-necessary nodes, after which they would search the DHT for
-nodes that are interested in `F` and gossip this proof to them..
+> For example, let this be the proof: `[A, B, C, D, E, F]`, then `A` is the root note and `F` is
+> the target node. This proof should be gossiped to nodes that are interested in storing node `F`.
 
-The recipients of that gossip are then responsible for gossiping the parent
-intermediate node `E`.  This process repeats until it terminates at the state
-root, with the final round of gossip only containing the `[A]` which is the
-state root node of the trie.
+The recipients of this gossip are then responsible for gossiping the penultimate node of the proof
+(parent of the target node, `E` in our example). To do so, they would strip the last (target) node
+from the proof, search the DHT for nodes that are interested in `E` and gossip this proof to them.
 
+This process repeats until it terminates at the state root, with the final round of gossip only
+containing the `[A]` which is the state root node of the trie.
 
```

Comment:
While I like this explanation more than the current one, maybe we should also explicitly mention that contract bytecode should be gossiped as-is i.e. there's no recursive process.

Reply:
I agree. I said it explicitly few lines below (consistent with other text).